instrumentation-api/build.gradle.kts

```diff
@@ -14,6 +14,7 @@ group = "io.opentelemetry.instrumentation"
 dependencies {
   api("io.opentelemetry:opentelemetry-api")
   implementation("io.opentelemetry:opentelemetry-extension-incubator")
+  implementation("io.opentelemetry:opentelemetry-semconv")
```

Comment:
oh, sorry, I see the problem with my suggestion, we don't want to rely on (unstable) semconv artifact from `instrumentation-api`

go ahead and revert this latest change, and let's wait for @mateuszrzeszutek. I think probably we'll want to introduce a ThreadDetailsAttributesExtractor.

Reply:
> I think probably we'll want to introduce a ThreadDetailsAttributesExtractor.

Yeah, that would be my preferred way to do this. (Also see #4641)

We could just manually add the thread attributes extractor everywhere at first, but I hope that at some point we'd have something like #6651 but for `AttributesExtrator`s -- that's a future thing though.

Reply:
yup, i didn't want to add a dependency initially. wasn't thinking about unstable though. i can revert

Reply:
Will create ThreadDetailsAttributesExtractor.